app/javascript/shared/components/__tests__/focusTrap.test.jsx

```diff
@@ -0,0 +1,29 @@
+import { h } from 'preact';
+import { render } from '@testing-library/preact';
+import '@testing-library/jest-dom';
+import userEvent from '@testing-library/user-event';
+
+import FocusTrap from '../focusTrap';
+
+describe('<FocusTrap />', () => {
+  it('should trap focus within crayon-modals', async () => {
+    const { getByTestId } = render(
+      <div>
+        <input type="text " />
+        <FocusTrap>
+          <div class="crayons-modal">
+            <input type="text" data-testid="firstInput" />
+            <input type="text" />
+          </div>
+        </FocusTrap>
+      </div>,
+    );
+
+    const firstInput = getByTestId('firstInput');
+    firstInput.focus();
```

Comment:
(nitpick/refactor) - Another nice feature we're getting with your changes is that the first input in the focus trap should automatically receive focus - we could amend this test to verify that too - e.g.

```suggestion
    await waitFor(() => expect(firstInput).toHaveFocus());
```

Reply:
refactor (blocking): I would put this as a blocker as the test is more valid the way @aitchiss proposes.